pkg/services/contexthandler/contexthandler.go

```diff
@@ -742,6 +742,32 @@ type authHTTPHeaderListContextKey struct{}
 
 var authHTTPHeaderListKey = authHTTPHeaderListContextKey{}
 
+func withAllAuthHTTPHeaders(ctx context.Context, cfg *setting.Cfg) context.Context {
+	list := AuthHTTPHeaderListFromContext(ctx)
+	if list == nil {
+		list = &AuthHTTPHeaderList{
+			Items: []string{},
+		}
+	}
+
+	list.Items = append(list.Items, "Authorization")
+
+	if cfg.JWTAuthEnabled && cfg.JWTAuthHeaderName != "" {
+		list.Items = append(list.Items, cfg.JWTAuthHeaderName)
```

Comment:
These can also be `Authorization` but I think it won't lead to side effects in the current implementation